Iframe browsing frontend (#3198)
quiltdata/quilt: docs/Catalog/Preview.md

The Quilt catalog renders previews of the following file types.
Whenever possible, Quilt streams the smallest possible subset of the data
needed to generate the preview.

Previews are supported for uncompressed files as well as for gzip archives (.gz).

## Plain text previews

Quilt can display any plaintext file format, including the following.

* Most programming languages, with syntax highlighting
  (.cpp, .json, .js, .py, .sh,  .sql, etc.)
* Biological file formats
  (.bed, .cef, .gff, .fasta, .fastq, .sam, .pdbqt, .vcf, etc.)
* Text files (.csv, .md, .readme, .tsv, .txt, etc.)

## Chemical structures

The Quilt catalog uses the [NGL Viewer library](https://github.com/nglviewer/ngl)
to render structures.
By default, v3000 Molfiles are converted to v2000 by the JavaScript client
for rendering.

The following file formats are supported:

* Mol files (.mol, .mol2, .sdf)
* .cif
* .ent
* .pdb

## Binary and special file format previews

* Excel (.xls, .xlsx)
* FCS Flow Cytometry files (.fcs)
* Images (.gif, .jpg, .png, .tif, .tiff, etc.)
* Media (.mp4, .webm, .flac, .m2t, .mp3, .mp4, .ogg, .ts, .tsa, .tsv, .wav)
* .ipynb (Jupyter and Voila dashboards)
* .parquet

## Advanced

> You can use visualizations inside iframes using any libraries you want.

Reviewer: ```suggestion
The Quilt catalog supports secure, custom Javascript-enabled visualizations and dashboards embedded inside iframes.
```

> It's up to you what JS library you'll import and use.

Reviewer: ```suggestion
It is at your discretion which JS library (or libraries) you wish to enable for your Amazon S3 bucket.
```

Author: It sounds like, there is some configuration per bucket where customer can enable JS libraries. But actually customers could write JS/HTML/CSS code with any `<script src="any-library">`  tags

Author: Maybe
> It is at your discretion which JS library (or libraries) you wish to import in your HTML file.

?

Reviewer: Yes - this is better, thanks. Updated suggestion

> To enable this permissive visualizations, you can tick the "Enable permissive

Reviewer: ```suggestion
To enable "permissive" visualizations, check the `Enable permissive
```

> HTML rendering" checkbox in [Bucket settings](/catalog/admin#buckets). Keep in

Reviewer: ```suggestion
HTML rendering` checkbox in [Bucket settings](/catalog/admin#buckets). Please note
```

> mind, that the process of rendering HTML file creates a short living session,
> that allows sharing files of the packages publicly.

Reviewer: ```suggestion
that rendering Javascript-enabled HTML files in your S3 bucket or Quilt package creates a short-lived 
**publicly available** session for the lifetime of the session, although all session traffic _remains encrypted_.
```

> So, when you open `report.html`, a new session is created, and we render
> `/short-living-session-id/report.html` in iframe. This let us fetch all relative
> media and scripts in that page: `./img.jpg` becomes
> `/short-living-session-id/img.jpg`, `script.js` becomes
> `/short-living-session-id/script.js`, etc. But, also makes other files in the
> same package available by those paths
> `/short-living-session-id/was-not-shared-explicitly.txt`, even if not referenced
> in `report.html`.

Reviewer: ```suggestion
### Example:
1. `report.html` is a file that includes a publicly available JS library and custom embedded script.
2. Opening `report.html` generates  a new session `temporary-session-id`.
3. The file is served from the iframe relative-path `/temporary-session-id/report.html`.
4. All relative media and scripts are rendered in the same iframe relative-path format:
    - `./img.jpg` is served as `/temporary-session-id/img.jpg`
    - `script.js` is served as `/temporary-session-id/script.js`

> **All files in the same package** are made temporarily publicly-available (for lifetime of the session) under `/temporary-session-id`, even if not explicitly referenced in `report.html`.
```

Author: I'm not sure about this:
> 3. The file is served from the iframe relative-path `/temporary-session-id/report.html`.
The path is not relative, it's absolute: ``https://file-server/temporary-session-id/report.html`. Well, `./report.html` is "relative" here
